integration-test/src/test/groovy/de/gesellix/docker/client/DockerStackComposeIntegrationSpec.groovy

```diff
@@ -0,0 +1,162 @@
+package de.gesellix.docker.client
+
+import de.gesellix.docker.client.stack.DeployConfigReader
+import de.gesellix.docker.client.stack.DeployStackOptions
+import groovy.util.logging.Slf4j
+import spock.lang.Requires
+import spock.lang.Specification
+
+@Slf4j
+@Requires({ LocalDocker.available() && LocalDocker.supportsStack() })
+class DockerStackComposeIntegrationSpec extends Specification {
+
+    static DockerClient dockerClient
+
+    File composeFile
+    String composeContent = '''
+
+version: "3.3"
+services:
+  service:
+    image: alpine:${IMAGE_VERSION}
+    command: sh
+    deploy:
+      endpoint_mode: vip
+      mode: replicated
+      replicas: 1
+      restart_policy:
+        condition: on-failure
+        delay: 5s
+        max_attempts: 3
+        window: 120s
+    environment:
+      SOME_VAR: ${SOME_VAR}
+    networks:
+      - "my-subnet"
+    volumes:
+      - type: volume
+        source: shm
+        target: /dev/shm
+
+networks:
+  my-subnet:
+
+volumes:
+  shm:
+    driver_opts:
+      type: tmpfs
+      device: tmpfs
+      o: size=128m
+'''
+
+    def "deploy a new stack with compose file"() {
+        given:
+        dockerClient.initSwarm(newSwarmConfig())
```

Comment:
There's an overloaded `initSwarm()` method available without any parameters. It internally initializes a default swarm config. Feel free to change to the simpler form, then you can remove the `newSwarmConfig()` method in this test. I'm aware that the other integration tests haven't been cleaned accordingly, yet ;-)

Reply:
Done